src/v/cluster/partition.cc

```diff
@@ -504,23 +504,25 @@ partition::timequery(storage::timequery_config cfg) {
         co_return co_await cloud_storage_timequery(cfg);
     }
 
+    const auto may_answer_from_cloud = fallback_mode{
```

Comment:
This is a big change in behavior which is not mentioned in the commit message or PR body.
Previously we called `bounds_timestamp` after a scheduling point inside the `local_timequery` method. And now we're calling this method first and then we're calling `local_timequery`. The `local_timequery` method returns a future. The code is awaiting the future so there is a scheduling point on line 516.\

The scheduling point invalidates the result of `bounds_timestamp` because new `archival_metadata_stm` batch could be applied when the execution of the fiber is suspended. So essentially, the `bounds_timestamp` is not guaranteed to return the same value after any `co_await` in the code. Inside the `local_timequery` the value could already be different.

Reply:
Thanks for this- re-evaluating changes.